src/Client.php

```diff
@@ -146,22 +146,17 @@ public function setRawDataStream($data, $enctype = null)
      */
     public function setMethod($method = HttpRequest::METHOD_GET)
     {
-        switch ($method) {
-            case HttpRequest::METHOD_GET:
-                $this->setRequestMethod(HttpRequest::METHOD_GET);
-                break;
-            case HttpRequest::METHOD_POST:
-                $this->setRequestMethod(HttpRequest::METHOD_POST);
-                break;
-            case HttpRequest::METHOD_PUT:
-                $this->setRequestMethod(HttpRequest::METHOD_PUT);
-                break;
-            case HttpRequest::METHOD_DELETE:
-                $this->setRequestMethod(HttpRequest::METHOD_DELETE);
-                break;
-            case HttpRequest::METHOD_HEAD:
-                $this->setRequestMethod(HttpRequest::METHOD_HEAD);
-                break;
+        $method = strtoupper($method);
```

Comment:
Please don't change behavior, keep original `$method` as is, just change `==` to `===` since it compares string